libsolidity/formal/SMTChecker.cpp

```diff
@@ -748,6 +748,13 @@ bool SMTChecker::visit(MemberAccess const& _memberAccess)
 		defineGlobalVariable(accessedName + "." + _memberAccess.memberName(), _memberAccess);
 		return false;
 	}
+	else if (accessType->category() == Type::Category::Enum)
+	{
+		auto enumType = dynamic_cast<EnumType const*>(accessType.get());
+		solAssert(enumType, "");
+		defineExpr(_memberAccess, enumType->memberValue(_memberAccess.memberName()));
```

Comment:
Still wondering what happens to a struct member access of type `Enum`. Will add a test.

Reply:
```
contract C
{
    enum D { Left, Right }
    struct S { uint x; D d; }
    function f(S memory s) internal pure {
        assert(s.d == D.Left);
    }
}
```
The PR breaks the code above:
```
Internal compiler error during compilation:
/home/leonardo/devel/solidity/libsolidity/ast/Types.cpp(2331): Throw in function unsigned int dev::solidity::EnumType::memberValue(const ASTString&) const
Dynamic exception type: boost::exception_detail::clone_impl<langutil::InternalCompilerError>
std::exception::what: Requested unknown enum value d
[dev::tag_comment*] = Requested unknown enum value d
```

Reply:
Fixed